### ISSUE TITLE
Training and inference context lengths

### ISSUE THREAD

**Issue by jxiong21029** [etched-ai/open-oasis#19]
Hello,

In order to use the open-source oasis model effectively, some necessary information is missing:
1. What is the maximum sequence length that the oasis500m model was trained on?
2. Was the model trained with masking strategies like sliding window attention and/or transformer-XL style recurrence?
3. How do you handle context length at inference time? Do you simply discard the oldest tokens in the KV cache after a maximum time horizon, or do you simply stop generating once the KV cache is maxed out?
4. In the provided `generate.py` script, the noise schedule seems to apply a uniform noise level to all context tokens, approximately min(current step noise level, 300), rather than e.g. the pyramid scheduler described in the original diffusion forcing paper. Was this noise schedule selected heuristically or was it tuned specifically for this project, and is it the same schedule used for the live demo?

Thanks!

**Comment by julian-q** [comment]
Thanks for your detailed questions @jxiong21029 

1. 32 frames
2. Just trained on 32 frame sequences
3. Yep we discard the tokens past the latest 32 frames!
4. We experimented with different noise schedules and chose what works best. It seems that using a constant noise level for the context works well when you want to fully denoise each new frame one at a time. (As opposed to some use cases of Diffusion Forcing where you progressively denoise future frames.

**Comment by huskydoge** [comment]
Hi @julian-q 

1. May I know if there is any overlap between items in the dataset during training? For example, could one item contain 32 frames while another shares the middle 16 frames? Or are there other types of overlap?  

2. Additionally, in the report, it states:  

   > We solved this by deploying dynamic noising, which adjusts inference-time noise on a schedule, injecting noise in the first diffusion forward passes to reduce error accumulation, and gradually removing noise in the later passes so the model can find and persist high-frequency details in previous frames for improved consistency. Since our model saw noise during training, it learned to successfully deal with noisy samples at inference.  

   However, after reviewing the codebase, I couldn't find any relevant implementation of dynamic noising (only `stabilization_level` and its corresponding logic). Did I miss something or misunderstand this part?  

3. I am also using the diffusion forcing objective and attempting to reproduce your work. I use the same architecture (both DiT and VAE) as in your codebase. I found that the model struggles with generalization, when using larger dataset,  gone through same training steps, it performs far worse than in the overfitting settings (I have attached an example generated by overfitting in AR mode with DDIM50, and another one from ckpt trained on larger scale of dataset, and also the gt_video). Doing overfitting well might suggest that my code logic is correct. I am curious about the scale of your dataset in terms of unique frames, if you could share about this information.

https://github.com/user-attachments/assets/49a601bc-4e96-4830-a9ef-74584bba7f1b

https://github.com/user-attachments/assets/15412153-3f59-4a9b-b9c7-759cf2edd399

https://github.com/user-attachments/assets/28d8684a-0441-4276-a0f2-52dd334afe1c

There are still several misalignment between oasis setting and mine, including prediction type (I used epsilon), history len (I just use three history len) and noise scheduling (instead of using sigmoid schedule, I applied cosine-simple noise schedule in [History-Guided Diffusion Forcing](https://github.com/kwsong0113/diffusion-forcing-transformer/blob/2afc54837d9bac0e278b0ea5436b0ee0973e1472/algorithms/dfot/diffusion/noise_schedule.py#L47), as I tested that this noise schedule will give lower loss and faster convergence in my settings.

I would greatly appreciate any guidance and insights you can provide. Thanks again for your amazing work!

If @jxiong21029 also has any insights to share, I’d love to hear them. Thanks in advance!

**Comment by jxiong21029** [comment]
@huskydoge It's hard to know what the issue is without more information. However, I will note that for generative models, if the amount of compute is held constant, multiple epochs on a small dataset can result in better generation quality than fewer epochs on a larger dataset. The most extreme example of this is [hyperfitting](https://arxiv.org/abs/2412.04318).

In your case, though, simply doing more epochs on the larger dataset, until you reach the same or a comparable training loss as you observed in your overfitting experiment, could be good enough.

**Comment by PoliteApps** [comment]
@jxiong21029 thanks for discussing this topic here! Did you manage to create a trainning script? Do you mind sharing? Thanks!
